src/lib.rs

```diff
@@ -67,25 +71,35 @@ impl Heap {
             Heap {
                 bottom: heap_bottom,
                 size: heap_size,
+                used: 0,
                 holes: HoleList::new(heap_bottom, heap_size),
             }
         }
     }
 
-    /// Allocates a chunk of the given size with the given alignment. Returns a pointer to the
-    /// beginning of that chunk if it was successful. Else it returns `None`.
-    /// This function scans the list of free memory blocks and uses the first block that is big
-    /// enough. The runtime is in O(n) where n is the number of free blocks, but it should be
-    /// reasonably fast for small allocations.
-    pub fn allocate_first_fit(&mut self, layout: Layout) -> Result<NonNull<u8>, ()> {
+    pub fn align_layout(&self, layout: Layout) -> Layout {
```

Comment:
I think this could be a function without a `&self` argument, given that it does not need access to `self`. Also, could you please either add some basic documentation for this function or make it private (so that we don't have undocumented functions in our public API)?

Reply:
Did both.